language/vm/vm_runtime/src/interpreter.rs

```diff
@@ -0,0 +1,696 @@
+    }
+
+    fn resolve_function(
+        &mut self,
+        module: &'txn LoadedModule,
+        idx: FunctionHandleIndex,
+    ) -> VMResult<FunctionRef<'txn>> {
+        self.module_cache
+            .resolve_function_ref(module, idx)?
+            .ok_or_else(|| VMStatus::new(StatusCode::LINKER_ERROR))
+    }
+
+    fn call_native(&mut self, function: FunctionRef<'txn>) -> VMResult<()> {
+        let module = function.module();
+        let module_id = module.self_id();
+        let function_name = function.name();
+        let native_function = match dispatch_native_function(&module_id, function_name) {
+            None => return Err(VMStatus::new(StatusCode::LINKER_ERROR)),
+            Some(native_function) => native_function,
+        };
+        if module_id == *EVENT_MODULE && function_name == EMIT_EVENT_NAME.as_ident_str() {
+            self.call_emit_event()
+        } else {
+            let mut arguments = VecDeque::new();
+            let expected_args = native_function.num_args();
+            if function.arg_count() != expected_args {
+                // Should not be possible due to bytecode verifier but this
+                // assertion is here to make sure
+                // the view the type checker had lines up with the
+                // execution of the native function
+                return Err(VMStatus::new(StatusCode::LINKER_ERROR));
+            }
+            for _ in 0..expected_args {
+                arguments.push_front(self.operand_stack.pop()?);
+            }
+            let (cost, return_values) = match (native_function.dispatch)(arguments) {
+                NativeReturnStatus::InvalidArguments => {
+                    return Err(VMStatus::new(StatusCode::INTERNAL_TYPE_ERROR));
+                }
+                NativeReturnStatus::Aborted { cost, error_code } => {
+                    /*
+                    self.gas_meter
+                        .consume_gas(GasUnits::new(cost), &self.execution_stack)?;
+                     */
+                    return Err(VMStatus::new(StatusCode::ABORTED).with_sub_status(error_code));
+                }
+                NativeReturnStatus::Success {
+                    cost,
+                    return_values,
+                } => (cost, return_values),
+            };
+            /*
+            self.gas_meter
+                .consume_gas(GasUnits::new(cost), &self.execution_stack)?;
+                */
+            for value in return_values {
+                self.operand_stack.push(value)?;
+            }
+            Ok(())
+        }
+    }
+
+    fn call_emit_event(&mut self) -> VMResult<()> {
+        let msg = self.operand_stack.pop_as::<ByteArray>()?;
+        let count = self.operand_stack.pop_as::<u64>()?;
+        let key = self.operand_stack.pop_as::<ByteArray>()?;
+        let guid = EventKey::try_from(key.as_bytes())
+            .map_err(|_| VMStatus::new(StatusCode::EVENT_KEY_MISMATCH))?;
+        // TODO:
```

Comment:
Sorry can you help me get rid of the first item in the TODOs here, it was already resolved.